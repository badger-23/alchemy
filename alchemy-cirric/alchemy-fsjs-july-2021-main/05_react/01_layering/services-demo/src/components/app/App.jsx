import React from 'react';
import ThisDay from '../../containers/ThisDay';

export default function App() {
  return <ThisDay />;
}
