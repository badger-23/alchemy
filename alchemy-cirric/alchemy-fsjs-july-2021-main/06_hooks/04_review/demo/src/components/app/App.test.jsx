import React from 'react';
import { fireEvent, render, screen } from '@testing-library/react';
import { rest } from 'msw';
import { setupServer } from 'msw/node';
import { MemoryRouter } from 'react-router-dom';
import listPageOne from '../../fixtures/list1.json';
import listPageTwo from '../../fixtures/list2.json';
import characterDetail from '../../fixtures/character.json';
import App from './App';

const server = setupServer(
  rest.get('https://rickandmortyapi.com/api/character/', (req, res, ctx) => {
    const page = req.url.searchParams.get('page');
    let response;

    if (page === '1') response = listPageOne;
    else if (page === '2') response = listPageTwo;

    return res(ctx.json(response));
  }),
  rest.get('https://rickandmortyapi.com/api/character/:id', (req, res, ctx) => {
    return res(ctx.json(characterDetail));
  })
);

describe('Rick and Morty App', () => {
  beforeAll(() => server.listen());
  afterAll(() => server.close());

  it('initially displays a paginated list of characters', async () => {
    render(
      <MemoryRouter>
        <App />
      </MemoryRouter>
    );

    screen.getByAltText('Loading Rick and Mortys...');

    const ul = await screen.findByRole('list', { name: 'characters' });
    expect(ul).not.toBeEmptyDOMElement();

    const characterFromPageOne = await screen.findByText('Rick Sanchez');
    expect(characterFromPageOne).toBeInTheDocument();

    const nextPageButton = screen.getByRole('button', {
      name: 'Go to next page',
    });

    fireEvent.click(nextPageButton);

    const characterFromPageTwo = await screen.findByText('Aqua Morty');
    expect(characterFromPageTwo).toBeInTheDocument();
  });

  it('displays a character detail page', async () => {
    const { container } = render(
      <MemoryRouter initialEntries={['/1']}>
        <App />
      </MemoryRouter>
    );

    screen.getByAltText('Loading Rick and Mortys...');

    const character = await screen.findByText('Rick Sanchez');
    expect(character).toBeInTheDocument();

    expect(container).toMatchSnapshot();
  });
});
