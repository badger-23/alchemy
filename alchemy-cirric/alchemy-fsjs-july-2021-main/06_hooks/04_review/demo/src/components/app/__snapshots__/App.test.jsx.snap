// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rick and Morty App displays a character detail page 1`] = `
<div>
  <a
    href="/"
  >
    <h2>
      Go Back
    </h2>
  </a>
  <figure
    aria-label="character"
    class="character"
  >
    <img
      alt="Rick Sanchez"
      class="characterImage"
      src="https://rickandmortyapi.com/api/character/avatar/1.jpeg"
    />
    <figcaption>
      <h3>
        Rick Sanchez
      </h3>
      <p>
        Human
      </p>
      <p>
        Status: 
        Alive
      </p>
      <p>
        Origin: 
        Earth (C-137)
      </p>
      <p>
        Current Location: 
        Earth (Replacement Dimension)
      </p>
    </figcaption>
  </figure>
</div>
`;
